oxidized-transformers/src/util/mod.rs

```diff
@@ -50,9 +52,24 @@ pub(crate) mod tests {
         }
     }
 
-    pub(crate) fn assert_tensor_eq<T>(a: impl IntoArrayD<T>, b: impl IntoArrayD<T>, epsilon: T)
-    where
-        T: AbsDiffEq<Epsilon = T> + Clone + Debug,
+    /// Check that two tensors are equal with the given absolute (`epsilon`)
+    /// and relative (`max_relative`) tolerances.
+    macro_rules! assert_tensor_eq {
+        ($lhs:expr, $rhs:expr $(, $opt:ident = $val:expr)*) => {
+            crate::util::tests::assert_tensor_eq_($lhs, $rhs, Relative::default()$(.$opt($val))*)
+        };
+        ($lhs:expr, $rhs:expr $(, $opt:ident = $val:expr)*,) => {
+            crate::util::tests::assert_tensor_eq_($lhs, $rhs, approx::Relative::default()$(.$opt($val))*)
+        };
```

Comment:
Why do we need a second definition with a trailing comma?

Reply:
In case you write

```
assert_tensor_eq!(
    lhs,
    rhs,
    epsilon = 1e-4,
)
```

(Note the trailing comma after epsilon.)